src/control/cmd/dmg/pool.go

```diff
@@ -68,11 +70,17 @@ type PoolCreateCmd struct {
 
 // Execute is run when PoolCreateCmd subcommand is activated
 func (cmd *PoolCreateCmd) Execute(args []string) error {
-	if cmd.Size != "" && (cmd.ScmSize != "" || cmd.NVMeSize != "") {
-		return errIncompatFlags("size", "scm-size", "nvme-size")
+	var errIncompatFlagsNb int
+	for _, value := range [3]bool{cmd.Size != "", cmd.ScmSize != "" || cmd.NVMeSize != "", cmd.All} {
```

Comment:
Huh. This is clever, haven't seen this construct before but it makes sense. I suggest renaming `errIncompatFlagsNb` to something more like `seenSizeFlags` to make the logic clearer.

Only other suggestion I might make here is that you might want to define the input test array separately with each test on a new line. Slightly more verbose, but will make adding new tests clear.

Reply:
Following tasks will be done:
- [ ] Refactoring `errIncompatFlagsNb` to `seenSizeFlags`
- [ ] Refactoring declare testing array (improvement of code readability and maintenance)